Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -99,21 +114,44 @@ int ConvertCommand::Main(const std::vector<std::string>& args)
   }
   const DiscIO::BlobType format = format_o.value();
 
-  // Open the volume now for inspection
-  std::unique_ptr<DiscIO::Volume> volume = DiscIO::CreateVolume(input_file_path);
-  if (!volume)
+  // Open the blob reader
+  std::unique_ptr<DiscIO::BlobReader> blob_reader = DiscIO::CreateBlobReader(input_file_path);
+  if (!blob_reader)
   {
-    std::cerr << "Error: Unable to open disc image" << std::endl;
+    std::cerr << "Error: The input file could not be opened." << std::endl;
     return 1;
   }
 
+  // Open the volume
+  std::unique_ptr<DiscIO::Volume> volume = DiscIO::CreateDisc(input_file_path);
```

Comment:
Should we expose the `std::unique_ptr<VolumeDisc> TryCreateDisc(std::unique_ptr<BlobReader>& reader)` function and use it here so we don't create redundant blob readers?

Reply:
That wouldn't work. `TryCreateDisc` moves the blob reader if the operation is successful, and we need to be able to keep using the blob reader after creating the volume.

Do you think adding a `GetBlobReader` function to the `Volume` class could make sense instead?

Reply:
Hm, but if I move the blob reader into `CreateDisc(std::unique_ptr<BlobReader>)`, the blob reader is gone if creating the volume fails... And we can't know if creating the volume will fail until we actually call it. So we would still need to expose `TryCreateDisc`, which I'm a little reluctant to do because its behavior of sometimes moving what you pass in is non-intuitive.

This problem is a bit tricker than I thought. Could we skip it for this PR?

Reply:
Yeah sure, it's not really a big deal, I just thought about it while reading the diff.